jsx/StaticDataTable.js

```diff
@@ -303,6 +303,54 @@ class StaticDataTable extends Component {
   getSortedRows() {
     const index = [];
 
+    // TODO: data column type check should probably be done once at init,
+    // to find which columns have mixed types.
+    // First pass to check if there is a mix of types in the values
+    const typesFound = [];
+    // is the current sorted column with mixed type?
+    let isMixedType = false;
+    // not the default column
+    if (this.state.SortColumn !== -1) {
+      // only push string or number types, rest is considered undefined
+      //
+      // break out of this loop once we encounter two mixed types:
+      // number and string inside the sorting column
+      for (let i = 0; i < this.props.Data.length; i += 1) {
```

Comment:
I would use `for of` over index-based iteration.

Reply:
Agreed, but I was trying to stay in line with the next loop for consistency.